### PR TITLE
[Temporarily Inactive] zephyr: alh: add ALH zephyr driver

### PR DESCRIPTION
**Pull request by abonislawski**
Rename dai funcs, add zephyr alh driver,
add get_stream_id to dai driver
Switch to ALH zephyr driver in zephyr build

based on SSP example https://github.com/juimonen/sof/commits/ssp-wip

### CODE REVIEW

File: src/drivers/intel/alh/alh-zephyr.c [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+                break;
+        case 24:
+                params->frame_fmt = SOF_IPC_FRAME_S24_4LE;
+                break;
+        case 32:
+                params->frame_fmt = SOF_IPC_FRAME_S32_LE;
+                break;
+        default:
+		dai_warn(dai,"%d not supported format", cfg->word_size);
+        }
```
Comment (lyakh) [inline src/drivers/intel/alh/alh-zephyr.c:98]: indentation - like in SSP

**Review by lgirdwood — CHANGES_REQUESTED** [review]
@juimonen @abonislawski we need to do the abstraction at the layer above for both DMA and DAIs. i.e.for DAIs like ALH and SSP we need to
1) rename src/audio/dai.c to src/audio/dai-xtos.c
2) create a new file called src/audio/dai.c that maps the SOF dai module to Zephyr DAI driver APIs
Likewise for the DMAs
This avoids all the alh-zephyr.c and ssp-zephyr.c (+more) wrappers - as we map/wrap at the layer above.

**Comment by juimonen** [comment]
> @juimonen @abonislawski we need to do the abstraction at the layer above for both DMA and DAIs. i.e.for DAIs like ALH and SSP we need to
> 
> 1. rename src/audio/dai.c to src/audio/dai-xtos.c
> 2. create a new file called src/audio/dai.c that maps the SOF dai module to Zephyr DAI driver APIs
>    Likewise for the DMAs
>    This avoids all the alh-zephyr.c and ssp-zephyr.c (+more) wrappers - as we map/wrap at the layer above.

@lgirdwood I see we could unify ssp-zephyr and alh-zephyr etc. into a single wrapper file, but that's about it....??? the dai_probe etc name clash with zephyr needs to be resolved anyway.

**Review by kv2019i — CHANGES_REQUESTED** [review]
This modifies cAVS2.5 build which is probably not intentional. 

I'm guessing this is better done via https://github.com/thesofproject/sof/pull/5731 , right ?

**Comment by abonislawski** [comment]
Just to make sure everyone is up to date: this PR is using old way with zephyr wrappers in sof, it is not needed anymore as we are switching to native zephyr api in DAI PR where alh part from this PR is already included
or if the DAI PR will be merged without alh then this PR will just extend DAI PR with alh

**Comment by lgirdwood** [comment]
> Just to make sure everyone is up to date: this PR is using old way with zephyr wrappers in sof, it is not needed anymore as we are switching to native zephyr api in DAI PR where alh part from this PR is already included or if the DAI PR will be merged without alh then this PR will just extend DAI PR with alh

We will be using native Zephyr APIs in the same manner as the DMA API that has been merged so does this need closed ?

**Comment by abonislawski** [comment]
Yes, I will reopen if needed

**Pull request closed without merging**